addons/analytic/tests/test_analytic_account.py

```diff
@@ -189,3 +190,30 @@ def test_analytic_plan_account_child(self):
         plans_json = self.env['account.analytic.plan'].get_relevant_plans()
         self.assertEqual(2, len(plans_json),
                          "The parent plan should be available even if the analytic account is set on child of third generation")
+
+    def test_analytic_plan_account_parent(self):
+        """
+        Check that when assigning an analytic plan as the parent to a child analytic plan,
+        both plans must belong to the same company.
+        """
+        company_1, company_2 = self.env['res.company'].create([
+            {'name': 'company_1'},
+            {'name': 'company_2'}
+        ])
+        self.env.user.company_ids |= company_1 + company_2
+        parent_analytic_plan = self.env['account.analytic.plan'].create({
+            'name': 'Parent Plan',
+            'company_id': company_1.id,
+        })
+        child_analytic_plan_1 = self.env['account.analytic.plan'].create({
+            'name': 'Child Plan 1',
+            'company_id': company_1.id,
+            'parent_id': parent_analytic_plan.id,
+        })
+        self.assertEqual(child_analytic_plan_1.parent_id.id, parent_analytic_plan.id)
+        with self.assertRaises(UserError, msg='The parent and child analytic plans must belong to the same company'):
```

Comment:
How can this work?

Reply:
I've made some changes to the Test case.

Reply:
But why was the build green?

Reply:
Hello @william-andre 
This message is just for the understanding that Plans should have the same company,
it will replace the original one

Reply:
But why was the build green?

Reply:
@william-andre the message is just displayed when no `UserError` exception is raised in the `with` statement.

Reply:
Oh, my bad I was reading it as `assertRaisesRegex` :+1: